contrib/IECoreArnold/src/IECoreArnold/ParameterAlgo.cpp

```diff
@@ -166,15 +166,15 @@ void setParameterInternal( AtNode *node, AtString name, int parameterType, bool
 				if( const M44dData *data = runTimeCast<const M44dData>( value ) )
 				{
 					const Imath::M44f v( data->readable() );
-					AiNodeSetMatrix( node, name, const_cast<float (*)[4]>( v.x ) );
+					AiNodeSetMatrix( node, name, *reinterpret_cast<AtMatrix*>( const_cast<float(*)[4][4]>( &v.x ) ) );
```

Comment:
Do we really need all the nested casting? I don't suppose `reinterpret_cast<const AtMatrix &>( v )` work?

Reply:
I think at the time, I felt better about being more explicit about the reason this cast works is that the classes can be cast to float[4][4] - but you're right that this is pretty ugly.  I'll go ahead and change this